updater/lib/dependabot/updater/group_update_creation.rb

```diff
@@ -446,14 +467,20 @@ def deduce_updated_dependency(dependency, original_dependency)
         )
         dependency_snapshot.handled_dependencies << dependency.name
 
-        Dependabot::Dependency.new(
+        dependency_params = {
           name: dependency.name,
           version: dependency.version,
           previous_version: original_dependency.version,
           requirements: dependency.requirements,
           previous_requirements: original_dependency.requirements,
           package_manager: dependency.package_manager
-        )
+        }
+
+        if Dependabot::Experiments.enabled?(:dependency_has_directory)
+          dependency_params[:directory] = dependency.directory
+        end
+
+        Dependabot::Dependency.new(**dependency_params)
```

Comment:
ℹ️ Not a big deal, but we _could_ use the previous initialization pattern with

```
directory: dependency.directory if Dependabot::Experiments.enabled?(:dependency_has_directory)
```

Reply:
Will do

Reply:
Hmm both rubocop and sorbet are raising errors for this pattern ☹️

Reply:
I'm going to not follow this pattern for now since it's raising errors.